### PR TITLE
DCOS-21167: placement constraint show too early error

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
My idea was to refactor the constraint fields into one component which tracks the dirty state (so if the value has been changed). That way I think I could hide the error if there has not been any editing yet. 

**Checklist**
- [X] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: src/js/components/PlacementConstraintsField.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+import FormGroupHeadingContent
+  from "#SRC/js/components/form/FormGroupHeadingContent";
+
+import OperatorTypes from "#PLUGINS/services/src/js/constants/OperatorTypes";
+import PlacementConstraintsUtil
+  from "#PLUGINS/services/src/js/utils/PlacementConstraintsUtil";
+
+// &nbsp; to space the table rows
+const NBSP = "\u00A0";
```
Comment (weblancaster) [inline src/js/components/PlacementConstraintsField.js]: I think there is a util that returns NBSP

File: src/js/components/PlacementConstraintsPartial.js [hunk 2]
```diff
@@ -43,148 +22,30 @@ export default class PlacementConstraintsPartial extends Component {
...
 
   getPlacementConstraintsFields(data = []) {
-    const constraintsErrors = findNestedPropertyInObject(
-      this.props.errors,
-      "constraints"
-    );
     const hideValueColumn = data.every(function(constraint) {
       return PlacementConstraintsUtil.requiresEmptyValue(constraint.operator);
     });
+    debugger;
```
Comment (weblancaster) [inline src/js/components/PlacementConstraintsPartial.js]: forgot `debugger` here

File: src/js/components/PlacementConstraintsPartial.js [hunk 3]
```diff
@@ -43,148 +22,30 @@ export default class PlacementConstraintsPartial extends Component {
...
-            {!valueError &&
-              <FieldHelp>
-                {isLastField ? "A string, integer or regex value. " : NBSP}
-                {typeSettings &&
-                  !typeSettings.requiresValue &&
-                  !typeSettings.requiresEmptyValue &&
-                  "This field is optional."}
-              </FieldHelp>}
-          </FormGroup>
+      debugger;
```
Comment (weblancaster) [inline src/js/components/PlacementConstraintsPartial.js]: And here.

**Comment by DanielMSchmidt** [comment]
This is a duplicate of #2779

**Pull request closed without merging**